src/NexusMods.CLI/CliGuidedInstaller.cs

```diff
@@ -0,0 +1,215 @@
+using System.Diagnostics;
+using JetBrains.Annotations;
+using NexusMods.Abstractions.CLI;
+using NexusMods.Abstractions.CLI.DataOutputs;
+using NexusMods.Common;
+using NexusMods.Common.GuidedInstaller;
+
+namespace NexusMods.CLI;
+
+/// <summary>
+/// Implementation of an option selector for the CLI.
+/// </summary>
+[UsedImplicitly]
+public class CliGuidedInstaller : IGuidedInstaller
+{
+    private const string ReturnInput = "x";
+    private static readonly string[] TableOfOptionsHeaders = { "Key", "State", "Name", "Description" };
+    private static readonly object[] TableOfOptionsFooterBack = { ReturnInput, "", "Back", "" };
+    private static readonly string[] TableOfGroupsHeaders = { "Key", "Group" };
+    private static readonly object[] TableOfGroupsFooterContinue = { ReturnInput, "Continue" };
+    private static readonly object[] TableOfGroupsFooterFinish = { ReturnInput, "Finish" };
+
+    /// <summary>
+    /// The renderer to use for rendering the options.
+    /// </summary>
+    public IRenderer Renderer { get; set; } = null!;
+
+    /// <summary>
+    /// If true, all option selection will select the first choice.
+    /// </summary>
+    public bool SkipAll { get; set; }
+
+    /// <inheritdoc/>
+    public void SetupInstaller(string name) { }
+
+    /// <inheritdoc/>
+    public void CleanupInstaller() { }
+
+    /// <inheritdoc />
+    public Task<UserChoice> RequestUserChoice(GuidedInstallationStep installationStep, CancellationToken cancellationToken)
+    {
+        var currentGroup = installationStep.Groups.Length == 1
+            ? installationStep.Groups[0]
+            : null;
+
+        var selectedOptions = new List<SelectedOption>();
+
+        while (true)
+        {
+            RenderStep(installationStep, currentGroup, selectedOptions);
+            var input = SkipAll ? "0" : GetUserInput();
```

Comment:
I'd honestly prefer something that wasn't a number for this, like 's'.
`0` could be typed in by accident when trying to select the first option.